### ISSUE TITLE
`prefer-object-methods`

### ISSUE THREAD

**Issue by ljosberinn** [sindresorhus/eslint-plugin-unicorn#1172]
Similar to #1171. Sometimes, developers loop over objects in ways that could be simplified with more appropriate/modern Object methods. There are surely more examples here to be found, these are just from the top of my head.

## Fail

```js
// reducing where `Object.fromEntries` + `Object.entries` would be simpler
const foo = Object.keys(myObj).reduce((acc, key) => {
  const value = myObj[key];
  acc[key] = value + 1;
  return acc;
}, {})

// looping over keys to access value; should be `Object.values`
const foo = Object.keys(myObj).map((key) => {
  const value = myObj[key];
  return value + 1
})
```

## Pass

```js
const foo = Object.fromEntries(
  Object.entries(myObj).map(([key, value]) => [key, value + 1])
)

const foo = Object.values(myObj).map((value) => value + 1)
```

**Comment by ljosberinn** [comment]
closing in favor of #1066
